tests/platform_helper/test_command_copilot.py

```diff
@@ -1050,6 +1050,94 @@ def test_appconfig_ip_filter_policy_is_applied_to_each_service_by_default(self,
+        "dbt_platform_helper.commands.copilot.get_log_destination_arn",
+        new=Mock(
+            return_value='{"prod": "arn:cwl_log_destination_prod", "dev": "arn:dev_cwl_log_destination"}'
+        ),
+    )
+    @patch("dbt_platform_helper.utils.aws.get_aws_session_or_abort", new=Mock())
+    @mock_aws
+    def test_alb_validation_additional_address_list(self, fakefs):
+        """ALB validation should allow additional address list (optional)"""
+        fakefs.create_file(
+            EXTENSION_CONFIG_FILENAME,
+            contents="""
+    alb:
+      type: alb
+      environments:
+        development:
+          cdn_domains_list: 
+            test.domain.uktrade.digital: "test.domain.uktrade.digital"
+          additional_address_list: ["another.domain"]
+    """,
+        )
+        fakefs.add_real_file(FIXTURES_DIR / "valid_workspace.yml", False, "copilot/.workspace")
+
+        fakefs.create_file(
+            "./copilot/environments/development/manifest.yml",
+        )
+
+        fakefs.create_file(
+            "copilot/web/manifest.yml",
+            contents=" ".join([yaml.dump(yaml.safe_load(WEB_SERVICE_CONTENTS))]),
+        )
+
+        result = CliRunner().invoke(copilot, ["make-addons"])
+
+        assert "Wrong key 'additional_address_list'" not in result.output
+        assert result.exit_code == 0
+
+    @patch("dbt_platform_helper.jinja2_tags.version", new=Mock(return_value="v0.1-TEST"))
+    @patch(
+        "dbt_platform_helper.utils.versioning.running_as_installed_package",
+        new=Mock(return_value=False),
+    )
+    @patch(
+        "dbt_platform_helper.commands.copilot.get_log_destination_arn",
+        new=Mock(
+            return_value='{"prod": "arn:cwl_log_destination_prod", "dev": "arn:dev_cwl_log_destination"}'
+        ),
+    )
+    @patch("dbt_platform_helper.utils.aws.get_aws_session_or_abort", new=Mock())
+    @mock_aws
+    def test_alb_validation_no_config_defined(self, fakefs):
+        """Prod should be allowed to be defined without a config specified."""
+        fakefs.create_file(
+            EXTENSION_CONFIG_FILENAME,
+            contents="""
+    alb:
+      type: alb
+      environments:
+        default:
+          cdn_domains_list: 
+            test.domain.uktrade.digital: "test.domain.uktrade.digital"
+        development:
+         # empty config
+    """,
+        )
+        fakefs.add_real_file(FIXTURES_DIR / "valid_workspace.yml", False, "copilot/.workspace")
+
+        fakefs.create_file(
+            "./copilot/environments/development/manifest.yml",
+        )
+
+        fakefs.create_file(
+            "copilot/web/manifest.yml",
+            contents=" ".join([yaml.dump(yaml.safe_load(WEB_SERVICE_CONTENTS))]),
+        )
+
+        result = CliRunner().invoke(copilot, ["make-addons"])
+
+        assert "Key 'development' error" not in result.output
+        assert "None should be instance of 'dict'" not in result.output
+        assert result.exit_code == 0
```

Comment:
I approve of the additional test for this, but I'm wondering whether we should be asserting on the behaviour we are expecting here instead of the errors that triggered the adding of the new test/fix.

Perhaps we could paramaterise the happy path test and pass in a test case with an empty config?

Reply:
I think there's been a miscommunication on the second paragraph here. Probably my fault.

If you feel there should be distinct test cases, which I am inclined to agree with on second pass, there is no need to parameterise the tests as they only test one scenario each.

Reply:
Happy to discuss this.
We have a fix for the regression tests so this is still blocked while the fix gets reviewed + merged